File: content/en/04-llms/llms/_solutions/08_vision/08_vision.py
# %%
import chatlas
from pyhere import here

# %%
recipe_images = here("data/recipes/images/")
img_ziti = recipe_images / "ClassicBakedZiti.jpg"
img_mac_cheese = recipe_images / "CreamyCrockpotMacandCheese.jpg"

# %%
chat = chatlas.ChatPosit(model="zai-org/GLM-5.3-Flash")
chat.chat(
    "Give the food in this image a creative recipe title and description.",
    chatlas.content_image_file(img_ziti),
)

# %%
chat = chatlas.ChatPosit(model="zai-org/GLM-5.3-Flash")
chat.chat(
    "Write a recipe to make the food in this image.",
    chatlas.content_image_file(img_mac_cheese),
)

File: content/en/04-llms/llms/_solutions/08_vision/08_vision.R
library(ellmer)

recipe_images <- here::here("data/recipes/images")
img_pancakes <- file.path(recipe_images, "EasyBasicPancakes.jpg")
img_pad_thai <- file.path(recipe_images, "PadThai.jpg")

chat <- chat_posit(model = "zai-org/GLM-5.3-Flash")
chat$chat(
  "Give the food in this image a creative recipe title and description.",
  content_image_file(img_pancakes)
)

chat <- chat_posit(model = "zai-org/GLM-5.3-Flash")
chat$chat(
  "Write a recipe to make the food in this image.",
  content_image_file(img_pad_thai)
)
